### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by calebbourg**
This PR adds a new Pull Request Template

Please let me know if these sections make sense or if we should add anything else. I just used a template we use at my job as a reference. This can also evolve over time.

Right now It will look like the following:
## Description
_describe the intent of your changes here_

### Changes
* ...
* ...
* ...

### Testing Strategy
_describe how you or someone else can test and verify the changes_

### Concerns
_describe any concerns that might be worth mentioning or discussing_

### CODE REVIEW

**Review by naconnors — APPROVED** [review]
Kept it basic, I like it.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Description
+_describe the intent of your changes here_
+
+
+#### GitHub Issue: _link your GitHub issue here_
+
+### Changes
+- [ ] ...
+- [ ] ...
+- [ ] ...
```
Comment (dilyn-corner) [inline .github/pull_request_template.md:10]: Is there a reason to use - [ ] instead of * ?

**Review by jhodapp — APPROVED** [review]
LGTM

**Comment by dilyn-corner** [comment]
lgtm :)

**Pull request merged**